Build failure with MinGW on OS X when cross-compiling for Windows

Using [MinGW for OS X](http://crossgcc.rts-software.org/doku.php), I get the following error:

		my_project/build/i586-mingw32/lib/libmruby.a(io.o): In function `mrb_io_s_select':
		my_project/mrbgems/mruby-io/src/io.c:660: undefined reference to `select@20'
		my_project/mrbgems/mruby-io/src/io.c:660: undefined reference to `select@20'
		my_project/mrbgems/mruby-io/src/io.c:682: undefined reference to `__WSAFDIsSet@8'
		my_project/mrbgems/mruby-io/src/io.c:683: undefined reference to `__WSAFDIsSet@8'
		my_project/mrbgems/mruby-io/src/io.c:693: undefined reference to `__WSAFDIsSet@8'
		my_project/mrbgems/mruby-io/src/io.c:705: undefined reference to `__WSAFDIsSet@8'
		my_project/mrbgems/mruby-io/src/io.c:695: undefined reference to `__WSAFDIsSet@8'
		my_project/build/i586-mingw32/lib/libmruby.a(io.o):my_project/mrbgems/mruby-io/src/io.c:707: more undefined references to `__WSAFDIsSet@8' follow
		my_project/build/i586-mingw32/lib/libmruby.a(file.o): In function `mrb_file_basename':
		my_project/mrbgems/mruby-io/src/file.c:196: undefined reference to `sprintf_s'
		my_project/build/i586-mingw32/lib/libmruby.a(file.o): In function `mrb_file_dirname':
		my_project/mrbgems/mruby-io/src/file.c:169: undefined reference to `sprintf_s'
		collect2: error: ld returned 1 exit status
		my_project/build/i586-mingw32/lib/libmruby.a(io.o): In function `mrb_io_s_select':
		my_project/mrbgems/mruby-io/src/io.c:660: undefined reference to `select@20'
		my_project/mrbgems/mruby-io/src/io.c:660: undefined reference to `select@20'
		my_project/mrbgems/mruby-io/src/io.c:682: undefined reference to `__WSAFDIsSet@8'
		my_project/mrbgems/mruby-io/src/io.c:683: undefined reference to `__WSAFDIsSet@8'
		my_project/mrbgems/mruby-io/src/io.c:693: undefined reference to `__WSAFDIsSet@8'
		my_project/mrbgems/mruby-io/src/io.c:705: undefined reference to `__WSAFDIsSet@8'
		my_project/mrbgems/mruby-io/src/io.c:695: undefined reference to `__WSAFDIsSet@8'
		my_project/build/i586-mingw32/lib/libmruby.a(io.o):my_project/mrbgems/mruby-io/src/io.c:707: more undefined references to `__WSAFDIsSet@8' follow
		my_project/build/i586-mingw32/lib/libmruby.a(file.o): In function `mrb_file_basename':
		my_project/mrbgems/mruby-io/src/file.c:196: undefined reference to `sprintf_s'
		my_project/build/i586-mingw32/lib/libmruby.a(file.o): In function `mrb_file_dirname':
		my_project/mrbgems/mruby-io/src/file.c:169: undefined reference to `sprintf_s'
		collect2: error: ld returned 1 exit status

--- a/mrbgem.rake
+++ b/mrbgem.rake
@@ -10,5 +10,7 @@ MRuby::Gem::Specification.new('mruby-io') do |spec|
     #spec.cc.include_paths += ["C:/Windows/system/include"]
     spec.linker.library_paths += ["C:/Windows/system"]
   end
-
+  if build.kind_of?(MRuby::CrossBuild) && %w(x86_64-w64-mingw32 i686-w64-mingw32).include?(build.host_target)
+    spec.linker.libraries += ['ws2_32']
+  end
 end

